lib/CurrentUser.php

```diff
@@ -46,19 +40,30 @@ public function getUser(): ?IUser {
 	 * @return string
 	 */
 	public function getUserIdentifier() {
-		if ($this->identifier === null) {
-			$this->identifier = $this->getUID();
+		if ($this->identifier !== null) {
+			return $this->identifier;
+		}
+
+		$uid = $this->getUID();
+		if ($uid !== null) {
+			$this->identifier = $uid;
+			return $this->identifier;
+		}
 
-			if ($this->identifier === null) {
-				$this->identifier = $this->getCloudIDFromToken();
+		$cloudId = $this->getCloudIDFromToken();
+		if ($cloudId !== null) {
+			$this->identifier = $cloudId;
+			return $this->identifier;
+		}
 
-				if ($this->identifier === null) {
-					// Nothing worked, fallback to empty string
-					$this->identifier = '';
-				}
-			}
+		$nickname = $this->request->getHeader('X-NC-Nickname');
```

Comment:
We should validate this.

Reply:
Actually, not sure that we need any validation at all :thinking:

Reply:
Idk, where does this string end up and is it always escaped properly? If not, I'd at least remove special chars

Reply:
It will be displayed by vue, so safe, and stored in the DB, safe too.
And probably displayed in the activity mails. Might be an issue. I'll remove the special chars :+1: